client/state/admin-menu/reducer.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * Internal dependencies
+ */
+
+import { keyedReducer, withStorageKey } from 'state/utils';
+import 'state/data-layer/wpcom/admin-menu';
+import { ADMIN_MENU_RECEIVE } from 'state/action-types';
+
+export const adminMenu = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case ADMIN_MENU_RECEIVE:
+			return { ...state, ...action.menu };
+		default:
+			return state;
+	}
+};
+
+export default withStorageKey( 'adminMenu', adminMenu );
```

Comment:
Will admin-menus redux data be different per site?  If so, then we can use the [keyedReducer](https://github.com/Automattic/wp-calypso/blob/6a5b2472160c2439ed1c85d5dba58f1b87309eb1/client/state/README.md#L75) helper, which will require adding site_id to the actions.

Reply:
definitely per site, as we have plugins and site config affecting the menu

Reply:
Thanks. Yep im aware but do you agree that it isnt for this PR as the endpoint is still global and not site specific?

Reply:
Looking at the changes in D49005-code, the description says "We'll need to change this to be a site-specific endpoint", so I guess just something to keep in mind for later.

Reply:
Agreed, except not "in mind" but within an Issue which I will shortly create 😄

Reply:
Raised issue here https://github.com/Automattic/wp-calypso/issues/45511